splunk_connect_for_snmp_poller/manager/profile_matching.py

```diff
@@ -33,20 +36,28 @@ def extract_desc(realtime_collection):
     return sys_descr, sys_object_id
 
 
-def assign_profiles_to_device(profiles, device_desc):
+def assign_profiles_to_device(profiles, device_desc, host):
     result = []
     for profile in profiles:
-        if "patterns" in profiles[profile]:
-            match_profile_with_device(device_desc, profile, profiles, result)
+        if profiles[profile].get("patterns"):
```

Comment:
maybe 
`if profiles[profile] and profiles[profile].get("patterns"):`

just to be protected if somehow profiles[profile] will be None